tensorflow/python/ops/math_ops.py

```diff
@@ -1771,6 +1776,8 @@ def matmul(a,
           a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)
 
 
+_OverrideBinaryOperatorHelper(matmul, "matmul")
```

Comment:
Agreed.  In particular, it means that downstream libraries are free to call `__matmul__` without worrying about version grudginess.

Which makes me understand your `_MATMUL` function above. :)  Okay, that should stay.